### PR TITLE
EVG-2261 Merge test results in waterfall and task drawer

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: service/task_history.go [hunk 1]
```diff
@@ -315,6 +315,13 @@ func (uis *UIServer) taskHistoryPickaxe(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	for i := range last {
```
Comment (john-m-liu) [inline service/task_history.go]: you can use the second argument for range here (and in the other places)

File: service/waterfall.go [hunk 2]
```diff
@@ -362,6 +362,13 @@ func getVersionsAndVariants(skip, numVersionElements int, project *model.Project
 			return versionVariantData{}, errors.Wrap(err, "error fetching failed tasks")
 
 		}
+
+		for i := range failedAndStartedTasks {
+			if err := last[i].MergeNewTestResults(); err != nil {
```
Comment (john-m-liu) [inline service/waterfall.go]: this should be failedAndStartedTasks

File: service/task_history.go (hunk 1 above)
Comment (bsamek) [reply service/task_history.go]: I'm mutating the tasks inside the slice in place, so I don't want to use the second argument of range, which creates a copy.

File: service/waterfall.go (hunk 2 above)
Comment (bsamek) [reply service/waterfall.go]: done

**Comment by bsamek** [comment]
@tychoish The merge logic is well unit tested. We should, however, verify that this works in staging, especially since there may be a performance impact.

**Pull request merged**